### PR TITLE
[ticket/15880] Add core event display_user_activity_modify_sql

### PR DESCRIPTION
**Pull request by toxyy**
Allows the sql queries that fetch most active topic and forum to be modified.
These queries are only run when viewing a user's profile page.
You can exclude posts with certain parameters this way.

PHPBB3-15880

Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [x] Tests pass
- [ ] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15880

### CODE REVIEW

**Comment by toxyy** [comment]
!set Event

**Comment by toxyy** [comment]
Not sure if

```
$result = $db->sql_query_limit($db->sql_build_query('SELECT', $forum_sql_ary), 1);
$active_f_row = $db->sql_fetchrow($result);
$db->sql_freeresult($result);

$result = $db->sql_query_limit($db->sql_build_query('SELECT', $topic_sql_ary), 1);
$active_t_row = $db->sql_fetchrow($result);
$db->sql_freeresult($result);
```

is the best way to reformat this, I just copied the existing code and placed it adjacent to one another.

**Comment by 3D-I** [comment]
It's ok the reformatting.
But the code placement is not IMO.

Those two "fetches" should go exactly each one underneath its related query and the 2 sub-queries should go back to where they came from.

At this point you can dispatch those two arrays as you did but not into a new event, simply adding those two to the `core.display_user_activity_modify_actives` below. Hence get rid of your event and add those two arrays to it modifying the existing event.

At a first glance.

File: phpBB/includes/functions_display.php [hunk 1]
```diff
@@ -1196,17 +1196,54 @@ function display_user_activity(&$userdata_ary)
...
+		*
+		* @event core.display_user_activity_modify_sql
+		* @var	array	forum_sql_ary		SQL array to get the user's most active forum
+		* @var	array	topic_sql_ary		SQL array to get the user's most active topic
+		* @since 3.2.5-RC1
+		*/
+		$vars = array('forum_sql_ary', 'topic_sql_ary');
+		extract($phpbb_dispatcher->trigger_event('core.display_user_activity_modify_sql', compact($vars)));
+
+		$result = $db->sql_query_limit($db->sql_build_query('SELECT', $forum_sql_ary), 1);
```
Comment (3D-I) [inline phpBB/includes/functions_display.php]: It's ok the reformatting.
But the code placement is not IMO.

Those two "fetches" should go exactly each one underneath its related query and the 2 sub-queries should go back to where they came from.

At this point you can dispatch those two arrays as you did but not into a new event, simply adding those two to the `core.display_user_activity_modify_actives` below. Hence get rid of your event and add those two arrays to it modifying the existing event.

At a first glance.

**Comment by marc1706** [comment]
I've pondered a bit about whether this we should merge this. I understand that you wanted to prevent duplicating the two queries in your extension, however I think this might actually be the better solution than this event both from a maintenance viewpoint in phpBB and how much use this event would otherwise be.

Therefore I'm going to close this PR.

**Pull request closed without merging**